route/route.go

```diff
@@ -33,9 +34,22 @@ func New() *Router {
 	}
 }
 
+// WithInstrumentation returns a router with instrumentation support.
+func (r *Router) WithInstrumentation(instrh func(handlerName string, handler http.HandlerFunc) http.HandlerFunc) *Router {
+	return &Router{rtr: r.rtr, prefix: r.prefix, instrh: instrh}
+}
+
 // WithPrefix returns a router that prefixes all registered routes with prefix.
 func (r *Router) WithPrefix(prefix string) *Router {
-	return &Router{rtr: r.rtr, prefix: r.prefix + prefix}
+	return &Router{rtr: r.rtr, prefix: r.prefix + prefix, instrh: r.instrh}
+}
+
+// InstrumentHandler returns the handler wrapped by the instrumentation handler.
+func (r *Router) InstrumentHandler(handlerName string, handler http.HandlerFunc) http.HandlerFunc {
```

Comment:
This shouldn't be exposed in the API, instead use it within the class

Reply:
Do you mean wrapping with the instrumentation handler every time a route is registered (eg in `Get()`, `Post()`, ...) using `path` as the `handler` label value?

Reply:
Yes, that's one way.